test/PowerShellEditorServices.Test/Extensions/ExtensionCommandTests.cs

```diff
@@ -154,41 +151,33 @@ public async Task CanUnregisterCommand()
                 new BufferPosition(line: 1, column: 1),
                 BufferRange.None);
 
-            string commandName = "test.scriptblock";
-            string commandDisplayName = "ScriptBlock extension";
+            const string commandName = "test.scriptblock";
+            const string commandDisplayName = "ScriptBlock extension";
 
             EditorCommand removedCommand = null;
-            _extensionCommandService.CommandRemoved += (object sender, EditorCommand command) =>
-            {
-                removedCommand = command;
-            };
+            extensionCommandService.CommandRemoved += (object _, EditorCommand command) => removedCommand = command;
 
             // Add the command and wait for the add event
-            await _powershellContextService.ExecuteCommandAsync(new PSCommand()
-                .AddCommand("Register-EditorCommand")
-                .AddParameter("Name", commandName)
-                .AddParameter("DisplayName", commandDisplayName)
-                .AddParameter("ScriptBlock", ScriptBlock.Create("Write-Output \"Extension output!\"")));
+            await psesHost.ExecutePSCommandAsync(
+                new PSCommand()
+                    .AddCommand("Register-EditorCommand")
+                    .AddParameter("Name", commandName)
+                    .AddParameter("DisplayName", commandDisplayName)
+                    .AddParameter("ScriptBlock", ScriptBlock.Create("Write-Output \"Extension output!\"")),
+                CancellationToken.None).ConfigureAwait(true);
 
             // Remove the command and wait for the remove event
-            await _powershellContextService.ExecuteCommandAsync(new PSCommand()
-                .AddCommand("Unregister-EditorCommand")
-                .AddParameter("Name", commandName));
+            await psesHost.ExecutePSCommandAsync(
+                new PSCommand().AddCommand("Unregister-EditorCommand").AddParameter("Name", commandName),
+                CancellationToken.None).ConfigureAwait(true);
 
             Assert.NotNull(removedCommand);
             Assert.Equal(commandName, removedCommand.Name);
             Assert.Equal(commandDisplayName, removedCommand.DisplayName);
 
             // Ensure that the command has been unregistered
-            await Assert.ThrowsAsync<KeyNotFoundException>(() =>
-                _extensionCommandService.InvokeCommandAsync("test.scriptblock", editorContext));
-        }
-
-        public void Dispose()
-        {
-            _powershellContextService.Dispose();
+            _ = await Assert.ThrowsAsync<KeyNotFoundException>(
```

Comment:
Maybe `_ =` is not needed?
```suggestion
            await Assert.ThrowsAsync<KeyNotFoundException>(
```

Reply:
It is needed, there is a Roslyn code analyzer warning otherwise.

Reply:
Nvm, that's just the first suggested fix, after adding `ConfigureAwait(true)` the actual error goes away. Strange.